### PR TITLE
Fixed local mode actor id

### PR DESCRIPTION
**Pull request by justinwyang**
Addressed issue #5715 

I incremented task index in local mode, which is usually handled through a separate method call in non local mode. Originally, the unchanging task index ended up generating duplicate actor id's, which would end up overriding older actors, hence the error.

### CODE REVIEW

**Comment by justinwyang** [comment]
@richardliaw

**Comment by richardliaw** [comment]
cc @edoakes can you review this? feel free to deallocate

File: python/ray/actor.py [hunk 1]
```diff
@@ -390,6 +390,9 @@ def _remote(self,
         # Instead, instantiate the actor locally and add it to the worker's
         # dictionary
         if worker.mode == ray.LOCAL_MODE:
+            # Increment task_index, otherwise done via the worker.submit_task
+            # call in non local mode.
```
Comment (edoakes) [inline python/ray/actor.py]:
```suggestion
            # call in non-local mode.
```
Same for below

**Review by edoakes — COMMENTED** [review]
Please add a test to `test_local_mode` that checks this and should be good to go. Re-request when you've done that.

**Comment by justinwyang** [comment]
I accidentally added tests to a different file the first time, I've since force pushed a commit with changes to `test_local_mode`

**Comment by edoakes** [comment]
`test_local_mode` and the linter are both failing. Please fix those.

**Comment by justinwyang** [comment]
@edoakes In the original version, the incrementing of `task_index` happens in the `submit_task` function defined in `worker.py`, which I noticed you removed. Is this step still handled? (Since this is the core of the issue, it is likely why the tests I wrote are failing).

**Comment by edoakes** [comment]
Sorry about that, the code that this relates to is currently being updated quite a bit. I decided it's best to just generate a random actor ID when running in local mode to avoid issues like this in the future and took care of it because it required a cython method to generate the random IDs. Please see: https://github.com/ray-project/ray/pull/5863.

**Pull request closed without merging**